src/app/settings/shelf-settings/components/edit-shelf-action-panel-configuration/edit-shelf-action-panel-configuration.component.ts

```diff
@@ -51,4 +55,37 @@ export class EditShelfActionPanelConfigurationComponent implements OnInit {
   public cancel(): void {
     this.onCancel.emit()
   }
+
+  public get saveButtonErrorsTooltip(): string | undefined {
+    const invalidControlsLabels: string[] = this.findInvalidControlsLabels()
+    return invalidControlsLabels.length > 0 ? $localize`common.form-required-field.button-tooltip` + `: ${invalidControlsLabels.join(', ')}` : undefined
```

Comment:
A lot of stuff is happening in this ternary.
Ternaries are awesome and they should be used, but they are prone to become difficult to read quite easily.

Since you have so much going on in this one, I would suggest, that in this case, we simply use an if-statement for better readability:
```
if (invalidControlsLabels.length <= 0) {
  return
}
return $localize`common.form-required-field.button-tooltip` + `: ${invalidControlsLabels.join(', ')}`
```

Reply:
Fix